[2024-12-13T22:01:05.493+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: etl_premierleague_to_analisis.transfer_manuel_akanji_logs manual__2024-12-13T22:00:52.675197+00:00 [queued]>
[2024-12-13T22:01:05.649+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: etl_premierleague_to_analisis.transfer_manuel_akanji_logs manual__2024-12-13T22:00:52.675197+00:00 [queued]>
[2024-12-13T22:01:05.656+0000] {taskinstance.py:1308} INFO - Starting attempt 1 of 2
[2024-12-13T22:01:05.846+0000] {taskinstance.py:1327} INFO - Executing <Task(PythonOperator): transfer_manuel_akanji_logs> on 2024-12-13 22:00:52.675197+00:00
[2024-12-13T22:01:05.889+0000] {standard_task_runner.py:57} INFO - Started process 67897 to run task
[2024-12-13T22:01:06.037+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'etl_premierleague_to_analisis', 'transfer_manuel_akanji_logs', 'manual__2024-12-13T22:00:52.675197+00:00', '--job-id', '247', '--raw', '--subdir', 'DAGS_FOLDER/etl_premierleague_to_analisis.py', '--cfg-path', '/tmp/tmpj873k3wc']
[2024-12-13T22:01:06.105+0000] {standard_task_runner.py:85} INFO - Job 247: Subtask transfer_manuel_akanji_logs
[2024-12-13T22:01:07.758+0000] {task_command.py:410} INFO - Running <TaskInstance: etl_premierleague_to_analisis.transfer_manuel_akanji_logs manual__2024-12-13T22:00:52.675197+00:00 [running]> on host 653f470dc9b4
[2024-12-13T22:01:09.236+0000] {taskinstance.py:1545} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='etl_premierleague_to_analisis' AIRFLOW_CTX_TASK_ID='transfer_manuel_akanji_logs' AIRFLOW_CTX_EXECUTION_DATE='2024-12-13T22:00:52.675197+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-12-13T22:00:52.675197+00:00'
[2024-12-13T22:01:09.639+0000] {base.py:73} INFO - Using connection ID 'shared_postgres' for task execution.
[2024-12-13T22:01:09.954+0000] {sql.py:375} INFO - Running statement: SELECT unnamed_0, header_performance, header_expected, header_sca, header_passes, header_carries, header_take_ons, date, dayofweek, comp, round, venue, result, team, opponent, game_started, position, minutes, goals, assists, pens_made, pens_att, shots, shots_on_target, cards_yellow, cards_red, touches, tackles, interceptions, blocks, xg, npxg, xg_assist, sca, gca, passes_completed, passes, passes_pct, progressive_passes, carries, progressive_carries, take_ons, take_ons_won, match_report FROM premierleague.public.manuel_akanji_logs, parameters: None
[2024-12-13T22:01:09.993+0000] {sql.py:384} INFO - Rows affected: 68
[2024-12-13T22:01:10.177+0000] {base.py:73} INFO - Using connection ID 'shared_postgres' for task execution.
[2024-12-13T22:01:10.331+0000] {taskinstance.py:1824} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 181, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/operators/python.py", line 198, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/etl_premierleague_to_analisis.py", line 40, in transfer_table_in_same_connection
    db_hook.insert_rows(target_table, source_data, target_fields=columns, replace=True)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/common/sql/hooks/sql.py", line 465, in insert_rows
    sql = self._generate_insert_sql(table, values, target_fields, replace, **kwargs)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/postgres/hooks/postgres.py", line 299, in _generate_insert_sql
    raise ValueError("PostgreSQL ON CONFLICT upsert syntax requires an unique index")
ValueError: PostgreSQL ON CONFLICT upsert syntax requires an unique index
[2024-12-13T22:01:10.722+0000] {taskinstance.py:1345} INFO - Marking task as UP_FOR_RETRY. dag_id=etl_premierleague_to_analisis, task_id=transfer_manuel_akanji_logs, execution_date=20241213T220052, start_date=20241213T220105, end_date=20241213T220110
[2024-12-13T22:01:11.227+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 247 for task transfer_manuel_akanji_logs (PostgreSQL ON CONFLICT upsert syntax requires an unique index; 67897)
[2024-12-13T22:01:12.112+0000] {local_task_job_runner.py:225} INFO - Task exited with return code 1
[2024-12-13T22:01:12.841+0000] {taskinstance.py:2653} INFO - 0 downstream tasks scheduled from follow-on schedule check
